lib/gala/payment_token.rb

```diff
@@ -20,14 +21,15 @@ def initialize(token_attrs)
       self.transaction_id = headers["transactionId"]
       self.ephemeral_public_key = headers["ephemeralPublicKey"]
       self.public_key_hash = headers["publicKeyHash"]
+      self.application_data = headers["applicationData"]
     end
 
     def decrypt(certificate_pem, private_key_pem)
+      self.class.validate_signature(signature, ephemeral_public_key, data, transaction_id, application_data)
```

Comment:
Let's pull out signature validation from `decrypt` and make it it's own public method. Reason being: signature validation and decryption are two distinct steps that don't necessarily go hand in hand (as our internal use already shows). So, by extracting validation from this method, we give the caller the flexibility to determine when/if to run validation.